docs/text-predicates.rst

```diff
@@ -0,0 +1,217 @@
+==========================
+JanusGraph Text Predicates
+==========================
+
+JanusGraph provides APIs to query the data using Text Predicates. The types of Text Predicates available
+for querying depends on the Indexing backend used. While using Elasticsearch as Indexing backend exposes
+all types of Text Predicates, but using some other backend like Solr won't expose all the Text Predicates. 
+Hence, be careful with using the Text Predicates, and refer to original JanusGraph docs_.
+
+.. _docs: https://docs.janusgraph.org/latest/
+
+
+----------------------------
+Available Text Predicates
+----------------------------
+
+The Text Predicates are broadly divided into 2 categories as:
+    - Text search predicates which match against the individual words inside a text string after it has been tokenized. These predicates are not case sensitive.
+
+        - TextContains_
+        - TextContainsFuzzy_
+        - TextContainsPrefix_
+        - TextContainsRegex_
+
+    - String search predicates which match against the entire string value
+
+        - TextFuzzy_
+        - TextPrefix_
+        - TextRegex_
+
+^^^^^^^^^^^^^^^
+TextContains
+^^^^^^^^^^^^^^^
+
+This Predicate is part of *Text* package.
+TextContains is used to query for partial matches in values of properties. 
+The following example is based on GodsOfGraph which comes pre-packaged with JanusGraph.
+The query is to find all edges where ``reason`` contains ``breezes``
+
+
+.. code-block:: python
+
+    from janusgraph_python.core.attribute.TextPredicate.Text import Text
+
+    # Find all edges which contains the word breezes
+    edges = g.E().has("reason", Text.textContains("breezes")).next()
+    print(edges)
+
+    ==> e[55m-6hc-b2t-3bk][8400-lives->4304]
+
+    # Find all edges which contains the word breezes and loves
+    edges = g.E().has("reason", Text.textContains("breezes")).has("reason", Text.textContains("loves")).next()
+    print(edges)
+
+    ==> e[55m-6hc-b2t-3bk][8400-lives->4304]
+
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+TextContainsFuzzy
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This Predicate is part of *Text* package.
+TextContainsFuzzy is used to query for partial matches with Fuzzy logic in values of properties.
+The following example is based on GodsOfGraph which comes pre-packaged with JanusGraph.
```

Comment:
nit s/GodsOfGraph/Graph of the Gods

You can mention closer to the top that all of the examples are based on Graph of the Gods rather than repeating it in the docs for each predicate.